samza-test/src/test/java/org/apache/samza/test/processor/TestStreamProcessor.java

```diff
@@ -132,7 +132,7 @@ public void testStreamProcessorWithNoTask() {
     configMap.remove("task.class");
     final Config configs = new MapConfig(configMap);
 
-    StreamProcessor processor = new StreamProcessor(1, configs, new HashMap<>());
```

Comment:
+1 to adding a test. I disagree that it is optional though.

---

Again not necessarily related to this specific commit, but I would also test start / start, start / stop / stop, and stop without start.